packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -791,6 +800,26 @@ export function withExtendedResolver(
   );
 }
 
+function safeSwap(
+  from: string,
+  to: string,
+  current: string,
+  doResolve: StrictResolver
+): SourceFileResolution | undefined {
+  if (current.endsWith(from)) {
+    try {
+      const hmrModule = doResolve(to);
+      if (hmrModule.type === 'sourceFile') {
+        debug(`Using \`${to}\` implementation.`);
+        return hmrModule;
+      }
+    } catch {
+      // Fallback to the default `from` implementation.
+    }
```

Comment:
We shouldn't allow a fallback here or make more "noise" when this fails. After these changes, unless I'm missing something, we don't expect this to fail. We should complain loudly when this does